openstates/de/bills.py

```diff
@@ -1,693 +1,339 @@
-            motion = doc.xpath(xpath)
-            if motion:
-                break
-            # Will fail at validictory level if no motion found.
-
-        # Create the vote object.
-        vote = Vote(chamber, date, motion, passed,
-                    counts['yes_count'], counts['no_count'],
-                    counts['other_count'])
-
-        # Add source.
-        vote.add_source(url)
-
-        # Get the "vote type"
-        el = doc.xpath('//font[contains(., "Vote Type:")]')[0]
-        try:
-            vote_type = el.xpath('following-sibling::font[1]/text()')[0]
-        except IndexError:
-            vote_type = el.xpath('../following-sibling::font[1]/text()')[0]
-
-        vote['vote_type'] = vote_type
-
-        # Get an iterator like: name1, vote1, name2, vote2, ...
-        xpath = ("//font[re:match(., '^[A-Z]$')]"
-                 "/../../../descendant::td/font/text()")
-        data = doc.xpath(xpath, namespaces=namespaces)
-        data = filter(lambda s: s.strip(), data)
-
-        # Handle the rare case where not all names have corresponding
-        # text indicating vote value. See e.g. session 146 HB10.
-        data_len = len(data) / 2
-        tally = sum(v for (k, v) in vote.items() if '_count' in k)
-
-        if (0 < data_len) and ((data_len) != tally):
-            xpath = ("//font[re:match(., '^[A-Z]$')]/ancestor::table")
-            els = doc.xpath(xpath, namespaces=namespaces)[-1]
-            els = els.xpath('descendant::td')
-            data = [e.text_content().strip() for e in els]
-
-        data = iter(data)
-
-        # Add names and vote values.
-        vote_map = {
-            'Y': 'yes',
-            'N': 'no',
-            }
-
-        while True:
-
-            try:
-                name = data.next()
-                _vote = data.next()
-
-                # Evidently, the motion for vote can be rescinded before
-                # the vote is cast, perhaps due to a quorum failure.
-                # (See the Senate vote (1/26/2011) for HB 10 w/HA 1.) In
-                # this rare case, values in the vote col are whitespace. Skip.
-                if not _vote.strip():
+            title_text = tds[0].text_content().strip().lower()
+
+            if "primary sponsor" in title_text:
+                pri_sponsor_text = tds[1].text_content()
+                primary_sponsors = self.separate_names(pri_sponsor_text)
+                #sometimes additional sponsors are in a 3rd td
+                #other times the 3rd td contains a blank image
+                addl_sponsors = []
+                add_spons_text = tds[2].text_content().strip()
+                if add_spons_text:
+                    add_spons_text = add_spons_text.replace("Additional Sponsor(s):","")
+                    addl_sponsors = self.separate_names(add_spons_text)
+
+            if "co-sponsor" in title_text:
+                cosponsor_text = tds[1].text_content()
+                if "none" in cosponsor_text.lower():
```

Comment:
Conditional more specific to avoid weird names. But we're going to be in trouble when Joe Cosponsor gets elected somewhere.